portfolio/src/main/java/com/google/sps/servlets/AnalysisServlet.java

```diff
@@ -51,11 +51,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         video.loadCaptions();
         
         // Calculate the search query and sentiment.
-        float sentimentScore = new SentimentTools(video).getScore();
+        float sentimentScore;
+        try {
+            sentimentScore = new SentimentTools(video).getScore();
+        } catch (IOException e) {
+            sendErrorMessage(response, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Language service unavailable.");
+            return;
+        }
         SearchQuery searchQuery = new SearchQuery(video, SEARCH_QUERY_SIZE);
 
         // Make the param string and redirect the user to the results page.
         String paramString = "?q=" + searchQuery + "&score=" + sentimentScore + "&title=" + video.getTitle();
-        response.sendRedirect("results.html" + paramString);
+        try {
+            response.sendRedirect("/results.html" + paramString);
+            return;
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
This seems like a no-op?

Reply:
Fixed after some discussion on best practices with Nikhil. Added DEBUG flags to the servlets and printed based on these. Nikhil suggested this could be expanded in the future (he mentioned making a tangential logger class).